src/installDeps.ts

```diff
@@ -1,84 +1,185 @@
-import * as io from "@actions/io";
 import * as ghCore from "@actions/core";
 import * as path from "path";
+import { promises as fs } from "fs";
 import Crda from "./crda";
 import { Inputs } from "./generated/inputs-outputs";
 import { fileExists } from "./util/utils";
 
-const REQUIREMENTS_TXT = "requirements.txt";
-const POM_XML = "pom.xml";
+type DepsInstallType = "Go" | "Maven" | "Node.js" | "Pip" | "custom";
+
 const GO_MOD = "go.mod";
+const POM_XML = "pom.xml";
 const PACKAGE_JSON = "package.json";
+const PACKAGE_LOCK = "package-lock.json";
+const YARN_LOCK = "yarn.lock";
+const REQUIREMENTS_TXT = "requirements.txt";
 
 const ALL_MANIFESTS = [
-    REQUIREMENTS_TXT, POM_XML, GO_MOD, PACKAGE_JSON,
+    GO_MOD, POM_XML, PACKAGE_JSON, REQUIREMENTS_TXT,
 ];
 
-export async function installDeps(manifestPath: string): Promise<void> {
-    const lastSlashIndex = manifestPath.lastIndexOf("/");
-    const manifestFileName = manifestPath.slice(lastSlashIndex + 1);
-    let manifestDir = ".";
-    if (lastSlashIndex !== -1) {
-        manifestDir = manifestPath.slice(0, lastSlashIndex);
+/**
+ * @returns The resolved manifest path - the manifest path even if the input was empty.
+ */
+export async function findManifestAndInstallDeps(
+    manifestDirInput: string,
+    manifestFileInput: string,
+    depsInstallCmd: string[] | undefined
+): Promise<string> {
+    // ghCore.info(`${Inputs.manifest_directory} is ${manifestDir}`);
+    let manifestDir;
+    let manifestFilename;
+    let resolvedManifestPath;
+    let installType: DepsInstallType | undefined;
+
+    if (manifestFileInput) {
+        manifestDir = path.join(manifestDirInput, path.dirname(manifestFileInput));
```

Comment:
If I understand this correctly, `path.dirname` will not return anything in any case. As `manifestFileInput` will only contain file name.

Reply:
oops, that is a hold-over from beore